app/grandchallenge/profiles/views.py

```diff
@@ -197,3 +206,94 @@ def form_invalid(self, form):
         # and display an error message
         messages.add_message(self.request, messages.ERROR, "Incorrect token.")
         return response
+
+
+class SubscriptionUpdate(SuccessMessageMixin, UserPassesTestMixin, UpdateView):
+    model = UserProfile
+    form_class = SubscriptionPreferenceForm
+    template_name = "profiles/subscription_form.html"
+    success_message = "You successfully updated your subscription preferences."
```

Comment:
Could we add information in the success message about what they have changed their preferences to? Something like:

`"You successfully updated your email preferences. You will no longer receive newsletter/notification emails. You will continue to receive newsletter/notification emails."`